src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -10,14 +10,15 @@
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.ExternalAccess.VSTypeScript.Api;
 using Microsoft.VisualStudio.LanguageServices.Implementation.TaskList;
 using Microsoft.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.Telemetry;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
 {
     [Export(typeof(VisualStudioProjectFactory))]
-    internal sealed class VisualStudioProjectFactory
+    internal sealed class VisualStudioProjectFactory : IVsTypeScriptVisualStudioProjectFactory
```

Comment:
Rather than doing this, can we create a separate class in the VSTypeScript external access layer that implements IVsTypeScriptVisualStudioProjectFactory and wraps it? Or add [Export(typeof(IVsTypeScriptVisualStudioProjectFactory))] here so that way the TypeScript-side PR can just import that rather than import VisualStudioProjectFactory? The concern here being that this is still implicitly encoding an inheritence/implementation hierarchy for external access which is often the nastiest to untangle.